### PR TITLE
Make JUnitRunner Filterable

### PR DESCRIPTION
**Pull request by SergeKireev**
This would help solve https://github.com/pantsbuild/pants/issues/8594, and make this runner coherent with runners in junit libraries

### CODE REVIEW

File: src/main/scala/org/scalatestplus/junit/JUnitRunner.scala [hunk 1]
```diff
@@ -113,5 +114,10 @@ final class JUnitRunner(suiteClass: java.lang.Class[_ <: Suite]) extends org.jun
    *  @return the expected number of tests that will run when this suite is run
    */
   override def testCount() = suiteToRun.expectedTestCount(Filter())
+
+  override def filter(filter: TestFilter): Unit = {
+    if (filter.shouldRun(getDescription)) throw new NoTestsRemainException
```
Comment (jsirois) [inline src/main/scala/org/scalatestplus/junit/JUnitRunner.scala]: I think you need to negate the condition. Towards that end, is a test possible?

Comment (SergeKireev) [reply src/main/scala/org/scalatestplus/junit/JUnitRunner.scala]: You are right, my bad. I added a test to check the correct behavior

**Comment by cheeseng** [comment]
@SergeKireev Sorry for attending late to this, The changes looks great and I think it will be really helpful if we can merge this.

By any chance have you signed before Artima Inc. CLA:

http://www.artima.com/cla/contributorsLicenseAgreement.pdf

or for Corporate:

http://www.artima.com/cla/corpContributorsLicenseAgreement.pdf

Please kindly print, sign and scan back a copy to cla@artima.com, we apologize for any inconvenience caused, we intended to improve the process but somehow it just didn't happen yet.

Thanks again for your hard effort!

Cheers.

**Comment by plinlor** [comment]
@cla-bot[bot] check

**Review by bvenners — APPROVED** [review]

**Comment by cheeseng** [comment]
@SergeKireev Changes are merged in `main` branch, sorry for the super late respond.  🙏🏻

**Pull request closed without merging**